### PR TITLE
Better hover handling for Scrollbar

### LINKED ISSUE #81015: Scrollbar thumbs aren't selectable when semi-transparent.
When I tried to drag a scrollbar thumb on a desktop app (Linux), grabbing the thumb is pretty hard in the first place, but I was surprised that when I dragged the thumb, it seemed to go the opposite direction from what I expected (the thumb ran away from my mouse pointer), until I realized that the click was instead just going straight through the thumb and the click was being interpreted as a drag of the list contents instead.

Probably when we click on a thumb, it should grab the thumb regardless of whether it is transparent or not, so that it won't be interpreted as a drag of the list contents below it.

https://user-images.githubusercontent.com/8867023/115801150-ea82f080-a390-11eb-8da0-fe7989f2df96.mp4

Sample code from the video (not really specific to this code, however):

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Scroll Bug',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Scroll Bug'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: ListView(
          children: List<Widget>.generate(100, (int index) {
            return Text('Item $index');
          }),
        ),
      ),
    );
  }
}
```

</details>

cc @Piinks

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/81015

This makes the hit test area of the scrollbar larger for mouse driven **_hover_** events. We previously reduced the size in #77354 where clicking/dragging with the mouse caused issues when the hit test area was enlarged. This will make it easier to grab the scrollbar with the mouse pointer.

It will also make the scrollbar reappear if we know where it is (we've scrolled at least once) if it has partially or fully faded away. Without information on the scrollable, the Scrollbar cannot paint, so if it has not been informed then we don't know where it is to calculate the hit test. That aspect is covered in #80262

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/scrollbar.dart [hunk 1]
```diff
@@ -399,7 +399,7 @@ class _MaterialScrollbarState extends RawScrollbarState<_MaterialScrollbar> {
   void handleHover(PointerHoverEvent event) {
     super.handleHover(event);
     // Check if the position of the pointer falls over the painted scrollbar
-    if (isPointerOverScrollbar(event.position, event.kind)) {
+    if (isPointerOverScrollbar(event.position, event.kind, forHover: true)) {
```
Comment (Piinks) [inline packages/flutter/lib/src/material/scrollbar.dart:402]: I don't love `forHover`, but I needed a way to distinguish different mouse events - click/drag v. hover

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 2]
```diff
@@ -481,22 +481,35 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
...
-    // The scrollbar is not able to be hit when transparent.
+
+    final Rect interactiveRect = _trackRect ?? _thumbRect!;
+    final Rect paddedRect = interactiveRect.expandToInclude(
+      Rect.fromCircle(center: _thumbRect!.center, radius: _kMinInteractiveSize / 2),
+    );
+
+    // The scrollbar is not able to be hit when transparent - except when
+    // hovering with a mouse. This should bring the scrollbar into view so the
+    // mouse can interact with it.
```
Comment (gspencergoog) [inline packages/flutter/lib/src/widgets/scrollbar.dart:503]: Ooh, this is going to be so nice... :-)

File: packages/flutter/lib/src/material/scrollbar.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/scrollbar.dart:402]: It's not horrible.  I mean, that's what it's for.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**